### PR TITLE
Base in guided tour

### PR DESCRIPTION
**Pull request by yminsky**
I pushed through the use of Base in the guided tour. This involved explaining a bit more about polymorphic compare and the lack thereof in Base, as well as explaining how local opens work. I also included a description of Base, Core_kernel and Core and how they fit in. 

Maybe the weirdest thing in this code is Base's **. operator, which I'm not sure how happy I am about, all in. I've started off a discussion about that amongst the Base devs.

### CODE REVIEW

File: book/01-guided-tour.html [hunk 1]
```diff
@@ -29,24 +29,39 @@ <h1>A Guided Tour</h1>
...
-      only on Unix. <code>Core</code> includes a lot of useful
-      functionality, including an excellent command line parsing
-      library and modules that provide access to low-level operating
-      systems facilities from mutexes to sockets. We'll discuss some
-      of the additional functionality provided by <code>Core</code>
-      later in the book.</p>
+      <h1><code>Base</code>, <code>Core</code> and <code>Core_kernel</code></h1>
+
+      <p><code>Base</code> is one of a family of three standard library
+      replacements, each with different use-cases. Here's a quick
```
Comment (avsm) [inline book/01-guided-tour.html]: Note that they build one each other and aren't entirely standalone?  "Family" does imply that I guess.

File: book/01-guided-tour.html [hunk 2]
```diff
@@ -29,24 +29,39 @@ <h1>A Guided Tour</h1>
...
+	  <li><em><code>Base</code></em> is the most minimal of the
+	    three.  It's designed to be portable, lightweight, and
+	    stable.
+	  </li>
+
+	  <li><em><code>Core_kernel<code></em>
+	    extends <code>Base</code> with many new data structures
+	    and capabilities. It's still portable, but has many more
+	    dependencies and takes longer to build, and also increases
+	    the size of your resulting executbale by more. Also, it
```
Comment (avsm) [inline book/01-guided-tour.html]: typo : executbale

File: book/01-guided-tour.html [hunk 3]
```diff
@@ -29,24 +29,39 @@ <h1>A Guided Tour</h1>
...
-      later in the book.</p>
+      <h1><code>Base</code>, <code>Core</code> and <code>Core_kernel</code></h1>
+
+      <p><code>Base</code> is one of a family of three standard library
+      replacements, each with different use-cases. Here's a quick
+	summary.
+
+	<ul>
+	  <li><em><code>Base</code></em> is the most minimal of the
+	    three.  It's designed to be portable, lightweight, and
```
Comment (avsm) [inline book/01-guided-tour.html]: I think it's worth elaborating a little on 'stable' here. Core_kernel states that the APIs change faster than those of Base.  Just how stable do we expect the Base library to be?  Also "lightweight" doesn't tell us much here.  Perhaps a sentence describing concretely what Base offers in terms of modules (fast datastructures such as Hashtable, and common conventions for error handling and serialisation)

File: book/01-guided-tour.html [hunk 4]
```diff
@@ -29,24 +29,39 @@ <h1>A Guided Tour</h1>
...
+	    dependencies and takes longer to build, and also increases
+	    the size of your resulting executbale by more. Also, it
+	    contains some APIs that change faster than those
+	    in <code>Base</code>
+	  </li>
+
+	  <li><em><code>Core</code></em> is the most full-featured,
+	    extending <code>Core_kernel</code> with support for a variety of
+	    UNIX APIs, but only works on UNIX-like OSs, including Linux
+	    and Mac OS.
```
Comment (avsm) [inline book/01-guided-tour.html]: Mac OS is now macOS.  We can fix this with a regexp globally

File: book/01-guided-tour.html [hunk 5]
```diff
@@ -177,22 +189,47 @@ <h1>Functions and Type Inference</h1>
       <code>int -&gt; int</code> is a function type, in this case
       indicating a function that takes an <code>int</code> and
       returns an <code>int</code>. We can also write functions that
-      take multiple arguments. (Note that the following example
-      will not work if you haven't opened
-      <code>Core_kernel</code> as was suggested
-      earlier.)<idx>multi-argument
-      functions</idx><idx>functions/with multiple
-      arguments</idx></p>
+      take multiple arguments. (Reminder: Don't forget to open <code>Base</code>,
```
Comment (avsm) [inline book/01-guided-tour.html]: shouldn't the `open` also be within the <code> tag here?

File: book/01-guided-tour.html [hunk 6]
```diff
@@ -177,22 +189,47 @@ <h1>Functions and Type Inference</h1>
...
+	the <code>of_int</code> function contained in
+	the <code>Float</code> module. This is different from what you
+	might expect from an object-oriented language, where
+	dot-notation is typically used for accessing a method of an
+	object. Note that module names always start with a capital
+	letter.
+      </p>
+
+      <p>
+	Modules can also be opened, as we opened <code>Base</code>
```
Comment (avsm) [inline book/01-guided-tour.html]: "Modules can also be opened" doesn't actually tell us what happened as a result of executing the `open Base`.  Perhaps note that all of the values contained within the module are now available in the toplevel environment?

File: book/01-guided-tour.html (hunk 1 above)
Comment (yminsky) [reply book/01-guided-tour.html]: Adjusted to clarify.

File: book/01-guided-tour.html (hunk 3 above)
Comment (yminsky) [reply book/01-guided-tour.html]: Urgh. I'm not sure what to say about "stable" concretely. Indeed, there are a few API changes that are still in the works. Do you have an example of the kind of extra concreteness you want?

Beyond that, I'll add a little more meat to address the rest of your concerns.

File: book/01-guided-tour.html (hunk 2 above)
Comment (yminsky) [reply book/01-guided-tour.html]: Fixed.

File: book/01-guided-tour.html (hunk 4 above)
Comment (yminsky) [reply book/01-guided-tour.html]: Fixed.

File: book/01-guided-tour.html (hunk 5 above)
Comment (yminsky) [reply book/01-guided-tour.html]: Sure. Done.

File: book/01-guided-tour.html (hunk 6 above)
Comment (yminsky) [reply book/01-guided-tour.html]: I expanded the language there.

**Comment by yminsky** [comment]
Ready for you to take another look.

**Pull request merged**